src/VecSim/algorithms/hnsw/serialization.cpp

```diff
@@ -232,15 +229,7 @@ void HNSWIndexSerializer::loadIndex(const std::string &location) {
     input.close();
 }
 
-void HNSWIndexSerializer::reset(std::shared_ptr<hnswlib::HierarchicalNSW<float>> hnsw_index_) {
-    // Hold the allocator, so it won't be destroyed while VecSimBaseObject is released.
-    auto allocator = hnsw_index->getAllocator();
-    if (hnsw_index_ == nullptr) {
-        hnsw_index.reset();
-        return;
-    }
```

Comment:
Why did you remove this?

Reply:
Not needed when using simple pointers. Also, we must use a simple pointer because we interact with the index through the C API.

Reply:
Two options - using shared pointers instead (requires right use in shared pointers), or using as is "unsafe" and then we should document properly that the index might be freed